lib/MusicBrainz/Server/Data/Recording.pm

```diff
@@ -175,6 +175,39 @@ sub find_by_works
     $self->query_to_list_limited($query, [$work_ids], $limit, $offset);
 }
 
+sub find_release_group_appearances
+{
+    my ($self, @recordings) = @_;
+
+    my %id_to_recordings = object_to_ids(@recordings);
+    my @ids = keys %id_to_recordings;
+    return unless @ids; # nothing to do
+    my $query = "SELECT rec.gid AS recording_gid, rg.id, rg.gid, rg.name,
+                    count(release.gid) AS recording_release_count,
+                    rgm.release_count AS total_release_count
+                 FROM recording rec
+                     JOIN track ON track.recording = rec.id
+                     JOIN medium ON medium.id = track.medium
+                     JOIN release ON release.id = medium.release
+                     JOIN release_group rg ON rg.id = release.release_group
+                     JOIN release_group_meta rgm ON rgm.id = rg.id
+                 WHERE rec.id IN (" . placeholders(@ids) . ")
+                 GROUP BY rec.gid, rg.id, rg.gid, rg.name, rgm.release_count
+                 ORDER BY rec.gid, count(release.gid) DESC, rgm.release_count";
+
+    my %map;
+    for my $row (@{ $self->sql->select_list_of_hashes($query, @ids) }) {
+        $map{ $row->{recording_gid} } ||= [];
+        push @{ $map{ $row->{recording_gid} } }, {
+            countTotal         => $row->{total_release_count},
+            countWithRecording => $row->{recording_release_count},
```

Comment:
The longer names seem less confusing (countTotal of what). Although see my general review comment, 'cause I'm not sure what these even mean. :P